data_collections/csv_updater.py

```
@@ -0,0 +1,27 @@
+    """
+    Save items to a csv file
+    """
+    if data:
+        try:
+            # Ensure directory exists
+            os.makedirs(os.path.dirname(path_to_file), exist_ok=True)
+            data_frame = pd.DataFrame(data)
+            data_frame.to_csv(path_to_file, index=False)
+            print(f"Items Successfully saved to {path_to_file}")
+        except Exception as e:
+            raise RuntimeError(f"Failed to save data to CSV: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)